nullaway/src/test/java/com/uber/nullaway/NullAwaySerializationTest.java

```diff
@@ -828,6 +828,41 @@ public void errorSerializationTest() {
         .doTest();
   }
 
+  @Test
+  public void errorSerializationEscapeSpecialCharactersTest() {
+    // Input source lines for this test are not correctly formatted intentionally to make sure error
+    // serialization will not be affected by any existing white spaces in the source code.
+    SerializationTestHelper<ErrorDisplay> tester = new SerializationTestHelper<>(root);
+    tester
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:SerializeFixMetadata=true",
+                "-XepOpt:NullAway:FixSerializationConfigPath=" + configPath))
+        .addSourceLines(
+            "com/uber/Test.java",
+            "package com.uber;",
+            "public class Test {",
+            "   Object m = new Object();",
+            "   public void run() {",
+            "     // BUG: Diagnostic contains: passing @Nullable parameter 'm.hashCode()",
+            "     foo(m.hashCode() == 2 || m.toString().equals('\\t') ? \t\n\t\n new Object() : null);",
+            "   }",
+            "   public void foo(Object o) { }",
+            "}")
+        .setExpectedOutputs(
+            new ErrorDisplay(
+                "PASS_NULLABLE",
+                "passing @Nullable parameter 'm.hashCode() == 2 || m.toString().equals('\\\\t') ? \\t\\n\\t\\n new Object() : null'",
+                "com.uber.Test",
+                "run()"))
+        .setFactory(errorDisplayFactory)
+        .setOutputFileNameAndHeader(ERROR_FILE_NAME, ERROR_FILE_HEADER)
+        .doTest();
```

Comment:
Actually... is this test even testing the TSV serialization code? It seem to me that this is just comparing the `ErrorDisplay` objects themselves, but doesn't go through the process of serializing/deserializing them :/

Reply:
Would you please tell me if it is necessary to do everything in reverse and construct the actual message  to pass the test ?

Currently we are writing the output below in the file:
```
passing @Nullable parameter 'm.hashCode() == 2 || m.toString().equals('\\t') ? \t\n\t\n new Object() : null' where @NonNull is required
```

Which is exactly what we need to reconstruct the actual message, if we don't escape special characters, the output below will be written to the file and the test will fail:
```
passing @Nullable parameter 'm.hashCode() == 2 || m.toString().equals('\t') ? 	
	
 new Object() : null' where @NonNull is required	
 ```

Isn't this sufficient to test escaping special characters in serialization ?

Reply:
Ok. Never mind. This seems correct to me. I was missing the fact that `SerializationTestHelper.readActualOutputs` is actually parsing the strings back from the TSV. That means it's going through the TSV serialization code, so this is what you want. My bad!